BraintreeCore/src/main/java/com/braintreepayments/api/core/SdkComponent.kt

```diff
@@ -0,0 +1,51 @@
+package com.braintreepayments.api.core
+
+import android.content.Context
+import androidx.work.WorkManager
+
+/**
+ * Component class that is created when the BT SDK is launched. It contains dependencies that need to be injected that
+ * contain Context.
+ */
+internal class SdkComponent(
+    applicationContext: Context,
+) {
+    val analyticsDatabase: AnalyticsDatabase = AnalyticsDatabase.getInstance(applicationContext)
+    val workManager: WorkManager = WorkManager.getInstance(applicationContext)
+    val configurationCache: ConfigurationCache = ConfigurationCache.getInstance(applicationContext)
+
+    companion object {
+        private var instance: SdkComponent? = null
+
+        /**
+         * Creates and returns a new instance of [SdkComponent], or returns the existing instance.
+         */
+        fun create(applicationContext: Context): SdkComponent {
+            return instance ?: SdkComponent(applicationContext).also { sdkComponent ->
+                instance = sdkComponent
+            }
+        }
+
+        /**
+         * Returns the instance of [SdkComponent]
+         */
+        fun getInstance(): SdkComponent {
+            return checkNotNull(instance)
```

Comment:
If this ever is null what happens at that point?

Reply:
It would throw an `IllegalStateException`. However this should never happen since `create()` would be called before `getInstance()`, which would set the singleton instance.